src/main/java/org/apache/commons/compress/archivers/tar/TarArchiveInputStream.java

```diff
@@ -1106,35 +917,8 @@ public int compare(final TarArchiveStructSparse p, final TarArchiveStructSparse
             }
         }
 
-        if (sparseInputStreams.size() > 0) {
+        if (!sparseInputStreams.isEmpty()) {
             currentSparseInputStreamIndex = 0;
         }
     }
-
-    /**
-     * This is an inputstream that always return 0,
-     * this is used when reading the "holes" of a sparse file
-     */
-    private static class TarArchiveSparseZeroInputStream extends InputStream {
```

Comment:
I moved this class out of the `TarArchiveInputStream` because I also needed it in the `TarFile`. Otherwise I would had to introduce more duplicated code. I made the class package private to prevent accidental usage.
In my opinion before the changes done here it was reasonable to have this as inner class as this was only used here.